From 35a752b2479c12f3ea6aa4474fe19a1345bb41b9 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:04:31 -0400
Subject: [usb] uhci: compile debugfs conditional on CONFIG_DYNAMIC_DEBUG

Message-id: <1410887463-20674-200-git-send-email-dzickus@redhat.com>
Patchwork-id: 95159
O-Subject: [RHEL7 PATCH 199/591] uhci: compile debugfs conditional on CONFIG_DYNAMIC_DEBUG || DEBUG
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit 9bab24afba2eb663efbd7f681731f8f49d42c4e8
Author: Oliver Neukum <oneukum@suse.de>
Date:   Mon Nov 18 13:23:10 2013 +0100

    uhci: compile debugfs conditional on CONFIG_DYNAMIC_DEBUG || DEBUG

    This makes sure the header files are all there

    Signed-off-by: Oliver Neukum <oneukum@suse.de>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/host/uhci-debug.c b/drivers/usb/host/uhci-debug.c
index 8e239cd..3989f0a 100644
--- a/drivers/usb/host/uhci-debug.c
+++ b/drivers/usb/host/uhci-debug.c
@@ -20,7 +20,7 @@
 
 static struct dentry *uhci_debugfs_root;
 
-#ifdef DEBUG
+#if defined(DEBUG) || defined(CONFIG_DYNAMIC_DEBUG)
 
 /* Handle REALLY large printks so we don't overflow buffers */
 static void lprintk(char *buf)
@@ -635,7 +635,7 @@ static const struct file_operations uhci_debug_operations = {
 
 #endif /* CONFIG_DEBUG_FS */
 
-#else /* DEBUG */
+#else /* DEBUG || CONFIG_DYNAMIC_DEBUG*/
 
 static inline void lprintk(char *buf)
 {}
-- 
1.7.1